api_v2/src/leaves/leaves.service.ts

```
@@ -161,6 +173,361 @@ export class LeavesService {
+      },
+      select: {
+        projectLeaderId: true,
+      },
+    });
+
+    const projectLeaderIds = [
+      ...new Set(
+        projectLeaders
+          .map((pl) => pl.projectLeaderId)
+          .filter((id): id is number => id !== null),
+      ),
+    ];
+
+    // Check notifications for disapprovals
+    for (const leaderId of projectLeaderIds) {
+      const notification = await this.prisma.notification.findFirst({
+        where: {
+          relatedEntityId: leaveId,
+          recipientId: leaderId,
+        },
+        orderBy: {
+          createdAt: 'desc',
+        },
+      });
+
+      if (notification) {
+        try {
+          const parsedData =
+            typeof notification.data === 'string'
+              ? JSON.parse(notification.data)
+              : notification.data;
+
+          if (parsedData.Status === RequestStatusEnum.DISAPPROVED) {
+            return true;
+          }
+        } catch (e) {
+          // Continue checking other leaders
+        }
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Marks a project leader's notification as resolved (approved/disapproved)
+   * @param leaveId - The leave ID
+   * @param projectLeaderId - The project leader's user ID
+   * @param isApproved - Approval status
+   */
+  private async markProjectLeaderNotificationResolved(
+    leaveId: number,
+    projectLeaderId: number,
+    isApproved: boolean,
+  ): Promise<void> {
+    const notifications = await this.prisma.notification.findMany({
+      where: {
+        relatedEntityId: leaveId,
+        recipientId: projectLeaderId,
+        OR: [{ type: 'leave' }, { type: 'undertime' }],
+      },
+    });
+
+    for (const notification of notifications) {
+      try {
+        let parsedData: any = {};
+        try {
+          parsedData =
+            typeof notification.data === 'string'
+              ? JSON.parse(notification.data)
+              : notification.data;
+        } catch (e) {
+          continue;
```

Add error message

The code swallows errors silently, making debugging impossible if something fails (like malformed JSON or a database issue).

- Silent error swallowing in notification update methods: catch (e) { continue }
- Makes debugging difficult and hides potential issues
- Recommendation: Log errors or use proper error handling